### PR TITLE
Autoclean for giant nodes

### PR DESCRIPTION
**Pull request by rustyrussell**
Autoclean cannot handle giant nodes:
1. The node OOMs even trying to listforwards.
2. If it didn't, autoclean would crash trying to queue that many delete commands.

There are improvements across the board, but the real win comes from only listing 10,000 entries at a time.

### CODE REVIEW

File: plugins/autoclean.c
Comment (vincenzopalazzo) [inline plugins/autoclean.c:1]: We should remove also the docs

File: plugins/autoclean.c [hunk 1]
```diff
@@ -8,61 +8,193 @@
...
+static struct plugin *plugin;
+/* This is NULL if it's running now. */
+static struct plugin_timer *cleantimer;
+
+enum subsystem_type {
+	FORWARDS,
+	PAYS,
+	INVOICES,
+#define NUM_SUBSYSTEM_TYPES (INVOICES + 1)
+};
```
Comment (vincenzopalazzo) [inline plugins/autoclean.c]: uh I did not know about this! looks amazing!

**Review by vincenzopalazzo — COMMENTED** [review]
Concept-ACK 

I left some minors comments

File: plugins/autoclean.c [hunk 2]
```diff
@@ -88,20 +91,23 @@ static const struct subsystem_ops subsystem_ops[NUM_SUBSYSTEM_TYPES] = {
 	{ {"succeededforwards", "failedforwards"},
 	  "listforwards",
 	  "forwards",
+	  "\"in_channel\":true,\"in_htlc_id\":true,\"resolved_time\":true,\"received_time\":true,\"status\":true",
```
Comment (niftynei) [inline plugins/autoclean.c:95]: oh you weren't joking about hacky!

File: plugins/libplugin.h [hunk 3]
```diff
@@ -455,6 +455,9 @@ bool plugin_developer_mode(const struct plugin *plugin);
 #define plugin_option_dev(name, type, description, set, arg) \
 	plugin_option_((name), (type), (description), (set), (arg), true, NULL, NULL, false)
 
+#define plugin_option_dev_dynamic(name, type, description, set, arg) \
```
Comment (niftynei) [inline plugins/libplugin.h:458]: nit: should be its own commit?

**Review by niftynei — COMMENTED** [review]
This was really fun to read through.

Notably, the counter is now off in `test_autoclean` (predates this change).

```
>       assert l3.rpc.autoclean_status()['autoclean']['expiredinvoices']['cleaned'] == 3
FAILED tests/test_plugin.py::test_autoclean - assert 5 == 3
```

Really nice usage of `filter` and the pagination APIs imo!!

**Comment by rustyrussell** [comment]
> This was really fun to read through.
> 
> Notably, the counter is now off in `test_autoclean` (predates this change).
> 
> ```
> >       assert l3.rpc.autoclean_status()['autoclean']['expiredinvoices']['cleaned'] == 3
> FAILED tests/test_plugin.py::test_autoclean - assert 5 == 3
> ```
> 
> Really nice usage of `filter` and the pagination APIs imo!!

This was actually a bug! (Yay for CI!) We weren't clearing the "done" counter and counting things twice...
